src/Types.hs

```diff
@@ -0,0 +1,36 @@
+-- Hoff -- A gatekeeper for your commits
+-- Copyright 2020 The Hoff authors
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- A copy of the License has been included in the root of the repository.
+
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Types
+(
+  PullRequestId (..),
+  Username (..),
+)
+where
+
+import Data.Aeson (FromJSON, ToJSON)
+import Data.String (IsString)
+import Data.Text (Text)
+import GHC.Generics (Generic)
+
+import qualified Data.Aeson as Aeson
+
+-- The name of a user on GitHub.
+newtype Username = Username Text deriving (Eq, Show, Generic, IsString)
+
+-- A pull request is identified by its number.
+newtype PullRequestId = PullRequestId Int deriving (Eq, Show, Generic)
+
+instance FromJSON PullRequestId
+instance FromJSON Username
```

Comment:
You could use newtype deriving instead of generic deriving, then you’ll get an error when someone changes the definition, rather than suddenly having an incompatible JSON encoding.